src/app/Routes.tsx

```diff
@@ -1,32 +1,81 @@
-import { FC } from 'react';
-import { Route, Routes as RRRoutes } from 'react-router-dom';
+import { FC, ReactElement } from 'react';
+import {
+  Navigate,
+  Route,
+  Routes as RRRoutes,
+  useLocation,
+} from 'react-router-dom';
 
 import Auth from '../features/auth/Auth';
 import Count from '../features/count/Counter';
-import Legacy from '../features/legacy/Legacy';
+import Legacy, { LEGACY_BASE_ROUTE } from '../features/legacy/Legacy';
 import Navigator from '../features/navigator/Navigator';
 import PageNotFound from '../features/layout/PageNotFound';
 import ProfileWrapper from '../features/profile/Profile';
+import { useAppSelector } from '../common/hooks';
+
+export const LOGIN_ROUTE = '/legacy/login';
+export const ROOT_REDIRECT_ROUTE = '/narratives';
 
 const Routes: FC = () => (
   <>
     <RRRoutes>
-      <Route path="/legacy/*" element={Legacy} />
+      <Route path={`${LEGACY_BASE_ROUTE}/*`} element={<Legacy />} />
       <Route
         path="/profile/:usernameRequested/narratives"
-        element={<ProfileWrapper />}
+        element={<Authed element={<ProfileWrapper />} />}
+      />
+      <Route
+        path="/profile/:usernameRequested"
+        element={<Authed element={<ProfileWrapper />} />}
+      />
+      <Route
+        path="/profile"
+        element={<Authed element={<ProfileWrapper />} />}
       />
-      <Route path="/profile/:usernameRequested" element={<ProfileWrapper />} />
-      <Route path="/profile" element={<ProfileWrapper />} />
-      <Route path="/count" element={<Count />} />
+      <Route path="/count" element={<Authed element={<Count />} />} />
       <Route path="/auth" element={<Auth />} />
-      <Route path={'/:id/:obj/:ver'} element={<Navigator />} />
-      <Route path={'/:category'} element={<Navigator />} />
-      <Route path={'/:category/:id/:obj/:ver'} element={<Navigator />} />
-      <Route path="/" element={<Navigator />} />
+      <Route
+        path={'/narratives/:id/:obj/:ver'}
+        element={<Authed element={<Navigator />} />}
+      />
+      <Route
+        path={'/narratives/:category'}
+        element={<Authed element={<Navigator />} />}
+      />
+      <Route
+        path={'/narratives/:category/:id/:obj/:ver'}
+        element={<Navigator />}
+      />
+      <Route path="/narratives" element={<Authed element={<Navigator />} />} />
+      <Route path="/" element={<HashRouteRedirect />} />
       <Route path="*" element={<PageNotFound />} />
     </RRRoutes>
   </>
 );
 
+export const Authed: FC<{ element: ReactElement }> = ({ element }) => {
```

Comment:
There might be a better name for this? This is the component wrapper that redirects users if they are unauthed, and is used above to declare authenticated routes.

Reply:
We discussed this together and decided that this name is fine.